### PR TITLE
feat(executor): Allow transient error when saving resource parameters

### PR DESCRIPTION
**Pull request by terrytangyuan**
We usually have long running tasks such as ML model training as resource type jobs (e.g. Kubeflow training CRDs) and we don't want the executor to fail due to transient errors at the very last phase when saving any of the parameters.

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

**Review by alexec — CHANGES_REQUESTED** [review]
Can you point me to the tests that verify this?

**Comment by terrytangyuan** [comment]
@alexec Yep, waiting for https://github.com/argoproj/argo-workflows/pull/5166 to merge and I'll probably do something similar to test an example. Although the transient error part may be non-trivial to test. This pattern is commonly seen throughout the codebase but no e2e test exists yet unless I miss anything. Any suggestions would be appreciated.

File: hack/test-examples.sh [hunk 1]
```diff
@@ -3,6 +3,9 @@ set -eu -o pipefail
 
 ./dist/argo delete -l workflows.argoproj.io/test
 
+# Grant admin privileges for the default service account so we could test the examples that submit k8s resources.
+kubectl create rolebinding default-admin --clusterrole=admin --serviceaccount=argo:default -n argo
```
Comment (terrytangyuan) [inline hack/test-examples.sh:7]: @alexec Is this okay? Or should we keep additional RBAC manifests that's more fine-grained in a separate folder?

File: workflow/executor/resource.go [hunk 2]
```diff
@@ -348,37 +349,48 @@ func (we *WorkflowExecutor) SaveResourceParameters(ctx context.Context, resource
...
-		out, err := cmd.Output()
-		if err != nil {
-			// We have a default value to use instead of returning an error
-			if param.ValueFrom.Default != nil {
-				out = []byte(param.ValueFrom.Default.String())
-			} else {
-				if exErr, ok := err.(*exec.ExitError); ok {
-					log.Errorf("`%s` stderr:\n%s", cmd.Args, string(exErr.Stderr))
+		var output string
+		err := waitutil.Backoff(ExecutorRetry, func() (bool, error) {
```
Comment (terrytangyuan) [inline workflow/executor/resource.go:372]: @alexec This pattern is commonly seen throughout the codebase and `waitutil.Backoff`+`argoerr.IsTransientErr` are covered in unit tests.  However there's no e2e test yet unless I miss anything. ` examples/k8s-jobs.yaml` exercises part of this code path but not the transient error part. Is this sufficient? Any suggestions would be appreciated.

**Comment by terrytangyuan** [comment]
Low priority as we only had this problem 3 times. Closing for now.

**Pull request closed without merging**